Islandora-CLAW/jsonld
# add hook_jsonld_alter_field_mappings

**GitHub Issue**: [Issue 916](https://github.com/Islandora-CLAW/CLAW/issues/916)

# What does this Pull Request do?

Creates a new hook, `hook_jsonld_alter_field_mappings`, allowing modules to add mappings for custom fieldtypes. 

# What's new?

A hook was added and the default fieldtype mappings were moved into the module's own implementation of the hook.

# How should this be tested?

- The module tests should run without errors. (Because I moved the fieldtype mappings to a hook implementation, the testJsonldcontextResponseIsValid test completing successfully shows that it works.)
- To really give it a run through:
    - Create a new module with a custom fieldtype
    - implement the new hook mapping the fieldtype
    - Add the field to a content type, give it an RDF mapping, and enable the json-ld serialization for that type
    - Create a new node/taxonomy_term using the new field
    - View the JSON-LD serialization for that node and make sure the mapping set by the hook is there.

# Additional Notes:

This doesn't actually resolve issue 916, which is a bit more complex. (I have a bit more testing to do before I issue the controlled_access_terms PR to fully resolve it.) However, supporting the serialization of custom fieldtypes via this method is still a good idea.

# Interested parties
@Islandora-CLAW/committers (esp @whikloj)
